File: Practica 5/Ejercicio 2/EstadisticasModular.java
import java.util.Scanner;
import java.util.Locale;

public class EstadisticasModular {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.useLocale(Locale.US); // Uso del punto como separador decimal

        System.out.println("Ingrese 10 números:");
        double[] numeros = new double[10];

        for (int i = 0; i < 10; i++) {
            if (scanner.hasNextDouble()) {
                numeros[i] = scanner.nextDouble();
            } else {
                System.out.println("Error: Entrada inválida, ingrese solo números.");
                scanner.next(); 
                i--; // Pedir de nuevo el número
            }
        }

        double promedio = calcularPromedio(numeros);
        double desviacion = calcularDesviacion(numeros, promedio);

        System.out.printf("El promedio es %.2f%n", promedio);
        System.out.printf("La desviación estándar es %.5f%n", desviacion);

        scanner.close();
    }

    public static double calcularPromedio(double[] datos) {
        double suma = 0;
        for (double num : datos) {
            suma += num;
        }
        return suma / datos.length;
    }

    public static double calcularDesviacion(double[] datos, double promedio) {
        double suma = 0;
        for (double num : datos) {
            suma += Math.pow(num - promedio, 2);
        }
        return Math.sqrt(suma / (datos.length - 1));
    }
}

File: Practica 5/Ejercicio 2/estadisticas_modular.py
import math

def promedio(numeros):
    return sum(numeros) / len(numeros)

def desviacion(numeros):
    prom = promedio(numeros)
    return math.sqrt(sum((x - prom) ** 2 for x in numeros) / (len(numeros) - 1))


numeros = list(map(float, input("Ingrese 10 números separados por espacio: ").split()))

print(f"El promedio es {promedio(numeros):.2f}")
print(f"La desviación estándar es {desviacion(numeros):.5f}")
